### DISCUSSION TITLE
I'm having an issue getting my application approved

### DISCUSSION THREAD

**Discussion by asc0910** [etsy/open-api#962] [Q&A]
We are a company that offers print-on-demand services to our clients, some of whom sell their artwork products on Etsy. We are looking to assist them by creating mockups or previews of their brand. Although we mentioned this feature in our application description, we received a rejection due to a "Generated API key at the request of 3rd party app". We believe that this does not violate the terms of service, as similar services are offered by Printful. Can anyone advise us on how to get approval?

**Comment by Gareth064** [comment]
First thing I would do is to email Etsy to ask what "Generated API key at the request of 3rd party app" actually means.

"Etsy Developer Support" <developer@etsy.com>

To me it sounds like they think you are going to ask users of your app to Generate their own API Key, give it to you to use in your app rather than you using your API key and then using the OAuth2 process with users to authenticate it against their shop. Doesn't make sense in that case.

But we need Etsy to tell us what these "reasons" actually mean.

Regarding advice on how to get approved, its a magically invisible set of rules that we have to figure out as we go along as Etsy don't give perfectly clear guidelines. The only piece of info is to tell them as much as you can about your app in the description box. But who knows if a human is reading these things or its some janky automated process.

**Reply by asc0910** [reply]
Thanks @Gareth064 for the quick answer.
Yes, I know the oauth2 flow to connect app to the shop and I have experience. I developed an Etsy app before and now I want to create a new app on another Etsy account but the submission was rejected.

**Reply by kev4ev** [reply]
I had the same situation a couple weeks back, same rejection reason. Because the email on my account was different than the app domain they had assumed a 3rd party was requesting me to generate credentials for them. 

A simple email to the address @Gareth064 mentioned above did the trick for me. I just explained that I am in fact the developer of the app and not requesting credentials on anyone else's behalf.
